pkg/receive/tsdb_test.go

```diff
@@ -16,74 +17,96 @@ import (
 	"github.com/prometheus/prometheus/pkg/labels"
 	"github.com/prometheus/prometheus/tsdb"
 
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/block"
 	"github.com/thanos-io/thanos/pkg/testutil"
 )
 
-func TestFlushableStorage(t *testing.T) {
+// TestHeadFlushIntoBlock ensures that the head flush
+// creates a block that is ready for shipping and
+// that flushing the storage does not cause data loss.
+// This test:
+// * opens a storage;
+// * appends values;
+// * flushes the storage wal/head; and
+// * queries the storage to ensure the samples are present from the flushed block.
+func TestHeadFlushIntoBlock(t *testing.T) {
```

Comment:
I don't think these unit tests really make sense any more. They used to ensure that our flushable storage implementation actually worked, but now we are essentially just replicating unit tests that should exist within prometheus. This is really testing foreign code; no Thanos packages/types are exercised here, right? Let's remove this IMO, otherwise we could do this is simply creating more maintenance overhead.

Reply:
done